examples/metrics/metrics/observer.js

```diff
@@ -14,29 +15,75 @@ const exporter = new PrometheusExporter(
 
 const meter = new MeterProvider({
   exporter,
-  interval: 2000,
+  interval: 3000,
 }).getMeter('example-observer');
 
-const otelCpuUsage = meter.createObserver('metric_observer', {
+meter.createObserver('cpu_core_usage', {
   monotonic: false,
-  labelKeys: ['pid', 'core'],
-  description: 'Example of a observer',
+  labelKeys: ['core'],
+  description: 'Example of a sync observer with callback',
+}, (observerResult) => {
+  observerResult.observe(getRandomValue(), { core: '1' });
+  observerResult.observe(getRandomValue(), { core: '2' });
 });
 
-function getCpuUsage() {
-  return Math.random();
-}
+const tempMetric = meter.createObserver('cpu_temp_per_app', {
+  monotonic: false,
+  labelKeys: ['app', 'core'],
+  description: 'Example of batch observer',
+});
+
+const cpuUsageMetric = meter.createObserver('cpu_usage_per_app', {
+  monotonic: false,
+  labelKeys: ['app', 'core'],
+  description: 'Example of batch observer',
+});
 
-const observable = new MetricObservable();
+meter.createBatchObserver('metric_batch_observer', (observerBatchResult) => {
+    function someAsyncMetrics() {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          const stats = [
+            {
+              name: 'app1',
+              core1: { usage: getRandomValue(), temp: getRandomValue() * 100 },
+              core2: { usage: getRandomValue(), temp: getRandomValue() * 100 },
+            },
+            {
+              name: 'app2',
+              core1: { usage: getRandomValue(), temp: getRandomValue() * 100 },
+              core2: { usage: getRandomValue(), temp: getRandomValue() * 100 },
+            },
+          ];
+          resolve(stats);
+        }, 200);
+      });
+    }
 
-setInterval(() => {
-  observable.next(getCpuUsage());
-}, 5000);
+    Promise.all([
+      someAsyncMetrics(),
+      // simulate waiting
+      new Promise((resolve, reject) => {
+        setTimeout(resolve, 300);
+      }),
+    ]).then(([apps, waiting]) => {
+      apps.forEach(app => {
+        observerBatchResult.observe({ app: app.name, core: '1' }, [
+          tempMetric.observation(app.core1.temp),
+          cpuUsageMetric.observation(app.core1.usage),
+        ]);
+        observerBatchResult.observe({ app: app.name, core: '2' }, [
+          tempMetric.observation(app.core2.temp),
+          cpuUsageMetric.observation(app.core2.usage),
+        ]);
+      });
+    });
+  }, {
+    maxTimeoutUpdateMS: 500,
```

Comment:
does the observer fail if the timeout is reached without any updates happening? What if some but not all happen? Does it throw or just log?

Reply:
the update itself is sync so once you start updating it will update all. Also there will be a debug message about timeout